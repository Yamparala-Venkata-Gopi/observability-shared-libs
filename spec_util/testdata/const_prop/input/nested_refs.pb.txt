# MethodTemplate proto

method_id: {
  name: "new_method_with_refs"
}

arg_templates: {
  key: "arg1"
  value: {
    ref: {
      method_index: 1
      arg_ref: {
        # Reference an arg that is itself a reference to a constant.
        key: "nested_ref_to_constant_arg"
        data_ref: {
          primitive_ref: {}
        }
      }
    }
  }
}

arg_templates: {
  key: "arg2"
  value: {
    ref: {
      method_index: 1
      arg_ref: {
        # Reference an arg that is itself a reference to a resposne.
        key: "nested_ref_to_response_arg"
        data_ref: {
          primitive_ref: {}
        }
      }
    }
  }
}
