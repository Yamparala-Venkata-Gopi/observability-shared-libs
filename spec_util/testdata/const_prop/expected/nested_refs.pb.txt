# MethodTemplate proto

method_id: {
  name: "new_method_with_refs"
}

arg_templates: {
  key: "arg1"
  value: {
    # This became a constant because the nested refs are resolved to a constant.
    value: {
      primitive: {
        type_hint: "string_constant_1"
        string_value: {
          value: "7c73565e-0d62-49b0-84e7-3063f4d53d9c"
        }
      }
    }
  }
}

arg_templates: {
  key: "arg2"
  value: {
    ref: {
      method_index: 1
      arg_ref: {
        # Reference an arg that is itself a reference to a resposne.
        key: "nested_ref_to_response_arg"
        data_ref: {
          primitive_ref: {}
        }
      }
    }
  }
}
